src/main/java/sim/coffee/Logger.java

```diff
@@ -29,29 +29,29 @@ public void add(Order[] o, String state) {
 			log.append(String.format("%-15s", "Processed"));
 			break;
 		}
-		
+
 		log.append(String.format("%-25s", "Customer ID: " + o[0].getCustomerID()));
 		log.append(String.format("%-7s", "Order: "));
 		for (Order order : o) {
-			log.append(order.getItemDetails());
+			log.append(order.getItemDetails() + " ");
 		}
 		log.append("\n");
 	}
-	
-	
+
+
 	//returns instance of the Log class
-	public static Log getInstance() {
+	public static Logger getInstance() {
 		if (instance == null)
-			instance = new Log();
+			instance = new Logger();
 		return instance;
 	}
-	
+
 	public StringBuilder getLog() {
 		return log;
 	}
-	
-	//writes log to file 
-	public void writeReport(String filename) {	
+
+	//writes log to file
+	public void writeReport(String filename) {
```

Comment:
Before we merge this PR we should probably set up the logic to make use of this method (I guess that'd be in manager, we need to figure out how to tell when all servers are done)

Reply:
For now I've implemented `writeReport()` in the Server class, the method is executed once the thread is finished (I realise this is not the most elegant solution and the file is written out at the finish of every thread which is not what we want). I'll still try to figure out how to call  `writeReport()` from the Manager class and if you think I should wait with the merge until then that's fair.
While testing `Log` methods I had trouble getting the `Server` threads running. I've pinned the problem down to the `!queue.isEmpty()` argument in the `while` loop (which could be due to an underlying flaw in the `SharedQueue` class and the way `empty` is set but I can't find for now). Removing the argument solves the problem of threads not running and looking at the log the last order still exits the queue and is processed despite `done` set to `true`. I've updated the `while` loop on this branch but not sure if we want to use this solution?

Reply:
@LinaRietuma Nice work and write up 😄 I can maybe take a look and see if I can figure that last bit out before we merge

Reply:
@kymckay Thanks 😄

Reply:
@LinaRietuma I fixed the logic in the loop 😄 Threw me for a loop too until it clicked that it was accidentally inverted (should be `||` not `&&`)

Reply:
@kymckay I think I came down to the same conclusion eventually but good to confirm 😄